synapse/storage/data_stores/main/devices.py

```diff
@@ -158,6 +187,16 @@ def get_devices_by_remote(self, destination, from_stream_id, limit):
                 # Stop processing updates
                 break
 
+            # skip over cross-signing keys
+            if (
+                update[0] in master_key_by_user
+                and update[1] == master_key_by_user[update[0]]["pubkey"]
```

Comment:
a comment about why it is correct to compare the device_id with the pubkey would be helpful... something like:

> cross-signing keys use the base64-encoded public key as the device id